mesonbuild/dependencies/pkgconfig.py

```diff
@@ -218,11 +219,17 @@ def _set_cargs(self) -> None:
             # so don't allow pkg-config to suppress -I flags for system paths
             env = os.environ.copy()
             env['PKG_CONFIG_ALLOW_SYSTEM_CFLAGS'] = '1'
-        ret, out, err = self._call_pkgbin(['--cflags', self.name], env=env)
+        ret, out, err = self._call_pkgbin(['--cflags-only-other', self.name], env=env)
         if ret != 0:
             raise DependencyException(f'Could not generate cargs for {self.name}:\n{err}\n')
         self.compile_args = self._convert_mingw_paths(self._split_args(out))
 
+        ret, out, err = self._call_pkgbin(['--cflags-only-I', self.name], env=env)
+        if ret != 0:
+            raise DependencyException(f'Could not generate cargs for {self.name}:\n{err}\n')
```

Comment:
Should cargs be changed for this exception or left as is? I could see both ways.